### PR TITLE
feat: lock app button, closes LEA-1543

### PR DESCRIPTION
**Pull request by edgarkhanzadian**
https://github.com/user-attachments/assets/7d3e85b4-0fde-4969-a846-0535c40ce62e

### CODE REVIEW

File: apps/mobile/src/components/splash-screen-guard/use-auth-state.tsx [hunk 1]
```diff
@@ -60,9 +60,17 @@ export function useAuthState({
     if (securityLevelPreference === 'secure') {
       setAnimationFinished(false);
       setAuthState('started');
-      userLeavesApp(+new Date());
+      // add latest active timestamp only if the app was actually unlocked
+      if (authState === 'passed-on-first' || authState === 'passed-afterwards')
```
Comment (pete-watters) [inline apps/mobile/src/components/splash-screen-guard/use-auth-state.tsx:64]: How about replacing the comment with a const? 

```suggestion
      const appUnlocked = authState === 'passed-on-first' || authState === 'passed-afterwards';
      if (appUnlocked)
```

**Review by pete-watters — APPROVED** [review]
👍

**Review by fbwoolf — APPROVED** [review]

File: apps/mobile/src/components/splash-screen-guard/use-auth-context.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { createContext, useContext } from 'react';
+
+interface AuthContextT {
```
Comment (kyranjamie) [inline apps/mobile/src/components/splash-screen-guard/use-auth-context.ts:3]:
```suggestion
interface AuthContextArgs {
```

File: apps/mobile/src/components/splash-screen-guard/use-auth-context.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  lockApp(): void;
+}
+
+export const AuthContext = createContext<AuthContextT>({
```
Comment (kyranjamie) [inline apps/mobile/src/components/splash-screen-guard/use-auth-context.ts:7]:
```suggestion
export const AuthContext = createContext<AuthContextArgs>({
```

not big on the T? That stands for type? Would avoid this in same way we don't use `IAuthContext` naming

**Pull request merged**